File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
class Solution {
    List<Integer> getElementsPresentInOneList(int[] nums1, int[] nums2) {
        Set<Integer> onlyInNums1 = new HashSet<> ();
        Set<Integer> existInNums2 = new HashSet<> ();
        
        for (int num : nums2) {
            existInNums2.add(num);
        }
        
        for (int num : nums1) {
            if (!existInNums2.contains(num)) {
                onlyInNums1.add(num);
            }
        }
        
        return new ArrayList<>(onlyInNums1);
    } 
    
    
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        return Arrays.asList(getElementsPresentInOneList(nums1, nums2), getElementsPresentInOneList(nums2, nums1));
    }
}

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        ans_0 = set(i for i in nums1 if i not in set(nums2))
        ans_1 = set(i for i in nums2 if i not in set(nums1))
        return [list(ans_0), list(ans_1)]
        